File: app/src/main/java/ru/magflayer/spectrum/data/database/dao/ColorNameDao.java
package ru.magflayer.spectrum.data.database.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

import ru.magflayer.spectrum.data.entity.ColorName;

@Dao
public interface ColorNameDao {

    @Query("SELECT * FROM ColorName")
    List<ColorName> loadColorNames();

    @Query("SELECT * FROM ColorName WHERE hex=:hex")
    ColorName loadColorNameByHex(String hex);

    @Query("SELECT COUNT(*) FROM ColorName")
    int getRowCount();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long[] saveColorNames(ColorName... colorNames);

}

File: app/src/main/java/ru/magflayer/spectrum/data/database/dao/ColorNameDao.kt
package ru.magflayer.spectrum.data.database.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import ru.magflayer.spectrum.data.entity.ColorName

@Dao
interface ColorNameDao {

    @get:Query("SELECT COUNT(*) FROM ColorName")
    val rowCount: Int

    @Query("SELECT * FROM ColorName")
    suspend fun loadColorNames(): List<ColorName>

    @Query("SELECT * FROM ColorName WHERE hex=:hex")
    suspend fun loadColorNameByHex(hex: String): ColorName

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun saveColorNames(vararg colorNames: ColorName): LongArray
}
